### ISSUE TITLE
Explicitly allow Multiple callings of update in one iteration

### ISSUE THREAD

**Issue by Libric0** [Libric0/loss_watch#6]
The current implementation of update has the problem that

1. Calling it multiple times leads to unexpected behaviour. For train losses, the train loss is appended, not just overwritten. For val losses, it is overwritten. This may be confusing
2. As long as train_loss is an obligatory parameter, you cannot just update for every val loss within their interval.

For the second point, this is the current state:
```python
for epoch, update in LossProgressBar(epochs):
    # Perform your models training step and retrieve a float `loss`
    train_loss = train_step()
    val_loss1 = None
    val_loss2 = None
    if epoch % 10 == 9:
        # A cheap validation
        val_loss1 = val_step()
    if epoch % 25 == 24:
        # A more expensive validation
        val_loss2 = val_step2()
    update(train_loss, val_step1=val_loss1, val_step2=val_loss2)
```

If we allow multiple callings within one iteration it could look like this:

```python
for epoch, update in LossProgressBar(epochs):
    # Perform your models training step and retrieve a float `loss`
    train_loss = train_step()
    update(train_loss)
    if epoch % 10 == 9:
        # A cheap validation
        val_loss1 = val_step()
        update(val_step1=val_loss1)
    if epoch % 25 == 24:
        # A more expensive validation
        val_loss2 = val_step2()
        update(val_step2=val_loss2)
```

I think this is by far more readable, and does not just add Nones in places where they don't belong. Here is the todo:
- [x] Fulfill #2, so that train_step becomes an **optional parameter**
- [x] ~for each loss, add a variable that marks the **last updated epoch**~ Just check whether the current epoch is a key.
- [x] If a loss is edited twice in the same epoch _with different values_, **give a warning**
